torchdynamo/optimizations/training.py

```diff
@@ -24,6 +24,51 @@
 log = logging.getLogger(__name__)
 
 
+def is_aot_autograd_safe_to_run(gm, example_inputs):
+    """
+    There are some known issues with Aot Autograd. This is a workaround to catch
+    such cases, and fallback to eager. We should fix these quickly.
+
+    Issues
+    1) LSTM - https://github.com/pytorch/torchdynamo/issues/1147
+    2) LSTM - https://github.com/pytorch/functorch/issues/586
+    3) TODO - Add an issue for set_grad_enabled
+    4) Input mutation - https://github.com/pytorch/torchdynamo/issues/1301
+    """
+    import functorch.compile
+
+    safe = True
+    # 1) LSTM module (tts_angular) - https://github.com/pytorch/functorch/issues/586
+    for submod in gm.modules():
+        if submod.__class__.__name__ == "LSTM":
+            safe = False
+
+    # 2) set_grad_enabled
+    has_set_grad_enabled = False
+    for node in gm.graph.nodes:
+        if node.target == torch._C._set_grad_enabled:
+            has_set_grad_enabled = True
```

Comment:
if this is already True, you can just return here, since this True marks it unsafe below, right?